### DISCUSSION TITLE
Additional options on bufferline or lualine example

### DISCUSSION THREAD

**Discussion by directormac** [LazyVim/LazyVim#925] [Q&A]
The Given example was 

```lua
{
  "nvim-lualine/lualine.nvim",
  event = "VeryLazy",
  opts = function(_, opts)
    table.insert(opts.sections.lualine_x, "😄")
  end,
},
```

I tried recreating it and putting my files on "plugins/lua", didnt seem to work i wonder why, i did all the 3 examples on bufferline also, none of the options seem to override the default given by lazyvim. which method is preferred to override the defaults in lazy way and which is correct approach on dealing with list or object directly.

```lua
opts = function(_, opts)
  vim.list_extend(opts.options, {
    show_buffer_close_icon = false,
    show_close_icon = false,
  })
   table.insert(opts.options, {
    show_close_icon = false,
    show_buffer_close_icon = false,
  })
   opts.options = {
    show_buffer_close_icon = false,
    show_close_icon = false,
  }
end,
```

But doing it with the keys where e.g i put <A-1> to go to the first absolute buffer it works and does not overwrite the <leader>dp

```lua
keys = {
-- stylua: ignore
        { "<A-1>", function() require("bufferline").go_to(1, true) end, desc = "Go to first buffer", },
}
```

The key one is working but not the options.

**Comment by jyuan0** [comment]
To change the LazyVim configuration defaults for a plugin, you should put your changes in a `*.lua` file in the `plugins` directory, e.g. `plugins/lualine.lua` (see: https://github.com/folke/lazy.nvim#-structuring-your-plugins).

So, for the given lualine example, you can add a file with the following:
```
return {
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      table.insert(opts.sections.lualine_x, "😄")
    end,
  },
}
```

For bufferline, you can add the following:
```
return {
  {
    "akinsho/bufferline.nvim",
    opts = function(_, opts)
      opts.options.show_buffer_close_icons = false
      opts.options.show_close_icon = false
    end,
  },
}
```

**Reply by jyuan0** [reply]
Whether you choose to do this in one file or multiple doesn't matter, what matters is that it's a Lua file in the `plugins` directory. Some people seem to prefer one file per plugin, some seem to prefer grouping related plugins (e.g., `code.lua`, `ui.lua`, etc.), and some seem to prefer a mixture of the two.

**Reply by directormac** [reply]
this worked
```lua
return {
  {
    "akinsho/bufferline.nvim",
    opts = function(_, opts)
      opts.option = {
            any options to overwrite
       }
    end,
  },
}
```

Thank you! Hacked it out like 2 hours ago.

**Comment by directormac** [comment]
This also worked for lualine.
```lua
{
  "nvim-lualine/lualine.nvim",
  event = "VeryLazy",
  opts = {
    options = {
      component_separators = "",
      section_separators = "",
    },
    sections = {
      lualine_a = {
        {
          "mode",
          fmt = function(str)
            return ""
          end,
        },
      },
      lualine_z = {},
    },
  },
},

```

**Reply by stefanadelbert** [reply]
This worked for me.
